### PR TITLE
Update maintainership of rhsm modules

### PR DESCRIPTION
**Pull request by ptoscano**
- remove Barnaby, Adrian, and Kevil from `redhat_subscription`, as they are no more working on `subscription-manager`
- create a new `team_rhsm` group to maintain `redhat_subscription`, `rhsm_release`, and `rhsm_repository` (all the modules related to `subscription-manager)`
- add myself and Chris to `team_rhsm`

### CODE REVIEW

**Comment by felixfontein** [comment]
@barnabycourt @alikins @kahowell can you confirm that you are OK with being removed? (If you don't reply in four weeks, I'll assume you are OK with it and merge this - if you notice this only later we can easily re-add you.)

**Comment by alikins** [comment]
Yes please remove me. Haven’t worked on this in years.

**Comment by barnabycourt** [comment]
@ptoscano Please add @cnsnyder in my place

**Comment by barnabycourt** [comment]
shipit

**Comment by felixfontein** [comment]
@cnsnyder can you please confirm in this PR that you are OK with being added? (You can simply approve the PR, or write  a coment, whatever suits you best.)

**Review by cnsnyder — APPROVED** [review]
LGTM! Thank you for the update!

**Review by kahowell — APPROVED** [review]

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@barnabycourt @alikins @kahowell thank you very much for maintaining the redhat_subscription module in the past!
@ptoscano @cnsnyder thank you very much for stepping up as maintainers!

**Pull request merged**
